File: test_single_quotes.py
#!/usr/bin/env python3
"""Test single quote repair."""

from json_repair import repair_json
import json

test_cases = [
    ('{"asdf":"asdf"}', "Already valid"),
    ('{"asdf":\'asdf\'}', "Single quote value"),
    ("{'asdf':'asdf'}", "Single quotes everywhere"),
    ("{'asdf':\"asdf\"}", "Single quote key"),
    ('[1, 2, 3,]', "Trailing comma"),
    ('{"a": [1,2,],}', "Both issues"),
]

print("Testing Python json_repair with single quotes:")
print("=" * 60)

for test, description in test_cases:
    result = repair_json(test)
    try:
        json.loads(result)
        status = "✓ Valid"
    except:
        status = "✗ Invalid"
    
    changed = "→" if result != test else "="
    print(f"{description:25s}: {test!r:25s} {changed} {result!r:25s} [{status}]")

File: src/bin/test_single_quotes.rs
use repair_json::repair_json_aws_smithy;

fn main() {
    let test_cases = vec![
        (r#"{"asdf":"asdf"}"#, "Already valid"),
        (r#"{"asdf":'asdf'}"#, "Single quote value"),
        (r#"{'asdf':'asdf'}"#, "Single quotes everywhere"),
        (r#"{'asdf':"asdf"}"#, "Single quote key"),
        (r#"[1, 2, 3,]"#, "Trailing comma"),
        (r#"{"a": [1,2,],}"#, "Both issues"),
    ];

    println!("Testing Rust AWS Smithy repair with single quotes:");
    println!("============================================================");

    for (test, description) in test_cases {
        match repair_json_aws_smithy(test.as_bytes()) {
            Ok(result) => {
                let result_str = String::from_utf8_lossy(&result);
                let status = if result_str == test { "=" } else { "->" };
                println!("{:<25}: {:<25} {} {:<25} [OK]", description, test, status, result_str);
            }
            Err(e) => {
                println!("{:<25}: {:<25} = {:<25} [Error: {}]", description, test, test, e);
            }
        }
    }
}
